design-documents/graph-ql/schema-development.md

```diff
@@ -0,0 +1,85 @@
+
+In 2015, the [GraphQL][] spec was [first published](https://graphql.github.io/graphql-spec/July2015/) along with the reference implementation in JavaScript. It should be no surprise then, that today, JavaScript holds the strongest ecosystem of [GraphQL][] libraries, with [4,833 related npm packages](https://www.npmjs.com/search?q=graphql) at the time of this writing (10/2019). As such, the JavaScript ecosystem is a solid and safe choice, moving forward.
+
+## GraphQL Server
+
+Within the JavaScript ecosystem, there are [a few server choices available](https://www.npmtrends.com/apollo-server-vs-koa-graphql-vs-prisma), but the most popular and mature choice is [Apollo Server][]. Optionally, [`graphql-yoga`](https://www.npmjs.com/search?q=graphql-yoga) uses [Apollo Server][] under the hood.
+
+## Code-First vs. SDL
+
+As [previously mentioned](#graphql-ecosystem), there are myriad JavaScript libraries available to assist in the creation of a [GraphQL][] server, but before we explore them, we should consider the pros & cons of defining our schema as code-first vs. SDL. Many articles [have been written](#references) about this topic and there are excellent arguments on both sides, but code-first appears to be the direction things are headed.
+
+In this document, we will explore the code-first approach.
+
+> Instead of increasing the complexity of GraphQL server development with more tools, we should strive for a simpler development model. Ideally, one that lets developers leverage the programming language they're already using – [this is the idea of code-first](https://www.prisma.io/blog/the-problems-of-schema-first-graphql-development-x1mn4cb0tyl3?source=post_page-----cf0e50d5ccff----------------------#code-first-a-language-idiomatic-way-for-graphql-server-development).
+
+### Pros
+
+- SDL is a generated artifact, not manually maintained.
+- The code-first schema is the single source of truth.
+   - No inconsistencies. Schema definition & resolvers are always in sync.
+   - No redundancy (duplication).
+   - Easier to maintain.
+- The shape of the API is closer to the data model.
+- Better IDE support & developer experience.
+- Better code reuse and type safety.
+- Can split by features/modules.
+- Terse (less boilerplate).
+- Declarative.
+- Scalable.
+
+### Cons
+
+- Some of [the latest code-first libraries](https://www.prisma.io/blog/introducing-graphql-nexus-code-first-graphql-server-development-ll6s1yy5cxl5) are not yet mature enough.
```

Comment:
We find Nexus to actually be quite stable, but the issue seems to be in prisma1/2 being unstable / missing features.

Reply:
Thanks for the feedback. Prisma definitely needs more work.

Reply:
I've requested for Prisma to download and install Magento so that they can add the Magento database schema to their internal list of schemas for testing, which is exactly what they've done. In fact, they already had a pretty large list as it was, so this was just one more.

The bottom line is that Prisma is determined to fix these kinks and add more MVP features before a GA release, hopefully in Q1 2020. I do think we should wait for that, but I'm also very optimistic about what they intend to deliver, from what I've already seen.